recsourcePayoutRefactor.jl

```diff
@@ -0,0 +1,83 @@
+using Printf
+
+# Province struct
+struct Province
+    name::String
+    money::Int
+end
+
+# Function to generate the power set of provinces
+function power_set(country::Vector{Province})
+    pow_list = Vector{Vector{Province}}([])
+    size = 2 ^ length(country)
+    push!(pow_list, Vector{Province}())
+
+    for prov in country
+        pow_set_size = length(pow_list)
+        for x in 1:pow_set_size
+            temp_vec = copy(pow_list[x])
+            push!(temp_vec, prov)
+            push!(pow_list, temp_vec)
+        end
+    end
+
+    # Remove the empty set
+    return pow_list[2:end]
+end
+
+# Function to print payoffs based on conversion rates
+function print_payoffs(power_set::Vector{Vector{Province}}, conversion_rates::Dict{Int,Int})

```

Comment:
Let's split this into a function that computes/returns the payoffs and another that prints. I think current format for payoffs now is a `Dict` of the list of players (a coalition basically) to their value.

Reply:
I just commented out the print function for simplicity's sake. I siphoned all the the results into a dictionary which will be used when I integrate your code with mine.